BRANCHNOTES.md

```diff
@@ -0,0 +1,456 @@
+```
+#include "..." search starts here:
+#include <...> search starts here:
+ /home/uli/.arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/1.22.0-80-g6c4433a-5.2.0/bin/../lib/gcc/xtensa-esp32-elf/5.2.0/../../../../xtensa-esp32-elf/include/c++/5.2.0
+ /home/uli/.arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/1.22.0-80-g6c4433a-5.2.0/bin/../lib/gcc/xtensa-esp32-elf/5.2.0/../../../../xtensa-esp32-elf/include/c++/5.2.0/xtensa-esp32-elf
+ /home/uli/.arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/1.22.0-80-g6c4433a-5.2.0/bin/../lib/gcc/xtensa-esp32-elf/5.2.0/../../../../xtensa-esp32-elf/include/c++/5.2.0/backward
+ /home/uli/.arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/1.22.0-80-g6c4433a-5.2.0/bin/../lib/gcc/xtensa-esp32-elf/5.2.0/include
+ /home/uli/.arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/1.22.0-80-g6c4433a-5.2.0/bin/../lib/gcc/xtensa-esp32-elf/5.2.0/include-fixed
+ /home/uli/.arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/1.22.0-80-g6c4433a-5.2.0/bin/../lib/gcc/xtensa-esp32-elf/5.2.0/../../../../xtensa-esp32-elf/include
+ /home/uli/.arduino15/packages/esp32/tools/xtensa-esp32-elf-gcc/1.22.0-80-g6c4433a-5.2.0/bin/../xtensa-esp32-elf/sysroot/usr/include
+End of search list.
+```
+for AVR:
+```
+#include "..." search starts here:
+#include <...> search starts here:
+ /home/uli/.arduino15/packages/arduino/tools/avr-gcc/7.3.0-atmel3.6.1-arduino5/bin/../lib/gcc/avr/7.3.0/include
+ /home/uli/.arduino15/packages/arduino/tools/avr-gcc/7.3.0-atmel3.6.1-arduino5/bin/../lib/gcc/avr/7.3.0/include-fixed
+ /home/uli/.arduino15/packages/arduino/tools/avr-gcc/7.3.0-atmel3.6.1-arduino5/bin/../lib/gcc/avr/7.3.0/../../../../avr/include
+End of search list.
+```
+As one can see with the ESP32-gcc not all include directories are named `include`. Parsing of this output is pretty trivial though.
+
+## `c_cpp_properties.json` Generator
+
+
+## Settings
+### Global Settings
+Under linux at `~/.config/Code/User/settings.json`, for instance:
+```json
+{
+    "arduino.additionalUrls": "",
+    "arduino.logLevel": "verbose",
+    "arduino.disableTestingOpen": true,
+    "workbench.editor.enablePreview": false
+}
+```
+Code: [src/arduino/arduinoSettings.ts](src/arduino/arduinoSettings.ts)  
+Code: [src/arduino/vscodeSettings.ts](src/arduino/vscodeSettings.ts)  
+Validator: [package.json](package.json)  
+
+### Project Settings
+Path in project `.vscode/arduino.json`
+```json
+{
+  "board": "arduino:avr:nano",
+  "configuration": "cpu=atmega328old",
+  "sketch": "examples/lcdpong-butenc/lcdpong-butenc.ino",
+  "port": "/dev/ttyUSB0"
+}
+```
+Code: [src/deviceContext.ts](src/deviceContext.ts)  
+Validator: [misc/arduinoValidator.json](misc/arduinoValidator.json)  
+
+## General Tasks
+### Removing existing Attempts which mess with c_cpp_properties.json or Intellisense
+
+Remove these as they are helpless attempts to get IntelliSense working:
+```ts
+//src/arduino/arduino.ts
+  tryToUpdateIncludePaths()
+  addLibPath(libraryPath: string)
```

Comment:
O_o

I'm a bit confused -- we're removeing addLibPath from the commands (see `package.json `), but it's still used internally a bunch, so we're not removing the implementation?

Does it's use internally (and in the UI ) still work?

(I'm just not really sure if more code should be removed, or what...)

Reply:
Ah, that's an oversight. Calling the `addLibPath` command from the UI doesn't make sense anymore, as the library paths will be added automatically, but `addLibPath` is still called vie the "Include Library" button in the Library Manager (see `LibraryItemView.tsx`).

So `addLibPath` should be renamed to `includeLibrary` as that's all it does now. As the library path will be detected and automatically added afterwards, there's no need for the "old" addLibPath anymore.